code/Test_definitions/checkDeviceSwap.feature

```diff
@@ -0,0 +1,213 @@
+        Given a valid phone number identified by the token or provided in the request body
+        And the device has been swapped more than 240 hours ago
+        When the request "checkDeviceSwap" is sent
+        Then the response status code is 200
+        And the value of response property "$.swapped" == false
+
+    @check_device_swap_5_out_of_max_age
+    Scenario: Check that the response shows that the device has not been swapped when the last swap was before the maxAge field
+        Given a valid phone number identified by the token or provided in the request body
+        And the last swap for this phone number's in the device is known
+        And the request body property "maxAge" is set to a value lower that the last known device swap
+        When the request "checkDeviceSwap" is sent
+        Then the response status code is 200
+        And the value of response property "$.swapped" == false
+
+    @check_device_swap_6_no_device_swap_no_max_age
+    Scenario: Check that the response shows that the device has not been swapped - maxAge is not provided in the request parameter
+        Given a valid phone number identified by the token or provided in the request body
+        And the device has never been swapped
+        When the request "checkDeviceSwap" is sent
+        Then the response status code is 200
+        And the value of response property "$.swapped" == false
+
+    @check_device_swap_7_no_device_swap_with_max_age
+    Scenario Outline: Check that the response shows that the device has never been swapped - maxAge is provided in the request
+        Given a valid phone number identified by the token or provided in the request body
+        And the device has never been swapped
+        And the request body property "maxAge" is set to a value equal or greater than "<hours>" within the allowed range
+        When the request "checkDeviceSwap" is sent
+        Then the response status code is 200
+        And the value of response property "$.swapped" == false
+
+        Examples:
+            | hours |
+            | 260   |
+            | 120   |
+            | 24    |
+            | 12    |
+
+    # This scenario may not apply if the OB implementation does not allow a valid access token to be issued for a phone number that has never been connected to the network and, therefore, cannot reach the API.
+    @check_device_swap_8_NoDeviceSwapPhoneNumberNeverConnected
+    Scenario: Check that the response shows that the device has not been swapped when the phone number has never connected to the Operators's network so the device has never been activated
+        Given I want to test "checkDeviceSwap" for a user who has a phone number that has never been activated in the network
+        And this phone number is provided in the request body
+        When the request "checkDeviceSwap" is sent
+        Then the response status code is 200
```

Comment:
We're not falling in this case: ` In case the phone number has never been installed in a device, or no data is available in the operators records (e.g. database error), API will return a 422 error`  ?